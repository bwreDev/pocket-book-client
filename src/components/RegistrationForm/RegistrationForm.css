.registration-button {
  background-color: #225a2a;
  color: white;
  border-radius: 15px;
  padding: 20px;
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 10px;
  width: 250px;
}

.registration-button:hover {
  background-color: lightgreen;
  color: #303030;
  cursor: pointer;
}

.RegistrationForm fieldset {
  border-style: none;
}

.RegistrationForm input {
  height: 40px;
  border-radius: 15px;
  border-style: none;
  margin: 5px;
  background-color: lightgreen;
}

@media only screen and (min-width: 600px) {
  .RegistrationForm input {
    margin-top: 20px;
    margin-bottom: 20px;
    height: 60px;
    font-size: 20px;
  }

  .RegistrationForm fieldset {
    height: 500px;
  }
}

@media only screen and (min-width: 1200px) {
  .RegistrationForm input {
    margin-top: 30px;
    margin-bottom: 30px;
    height: 80px;
    font-size: 24px;
  }

  .RegistrationForm fieldset {
    height: 750px;
  }
}
