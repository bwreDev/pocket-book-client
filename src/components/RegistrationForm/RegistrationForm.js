import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
import Navbar from '../../components/Navbar/Navbar';
import './RegistrationForm.css';

export default class RegistrationForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => {},
  };

  state = {
    error: null,
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const {
      first_name,
      last_name,
      username,
      password,
    } = event.target;

    this.setState({ error: null });
    AuthApiService.postUser({
      username: username.value,
      password: password.value,
      first_name: first_name.value,
      last_name: last_name.value,
    })
      .then((user) => {
        first_name.value = '';
        last_name.value = '';
        username.value = '';
        password.value = '';
        this.props.onRegistrationSuccess();
        this.props.history.push('/login');
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    return (
      <>
        <Navbar />
        <form
          className='RegistrationForm'
          onSubmit={this.handleSubmit}
        >
          <fieldset className='Registration-input'>
            <h3>Register</h3>
            <label htmlFor='RegistrationForm__first_name'></label>
            <input
              name='first_name'
              type='text'
              placeholder='First Name'
              required
              id='RegistrationForm__first_name'
            ></input>
            <label htmlFor='RegistrationForm__last_name'></label>
            <input
              name='last_name'
              type='text'
              placeholder='Last Name'
              required
              id='RegistrationForm__last_name'
            ></input>
            <label htmlFor='RegistrationForm__username'></label>
            <input
              name='username'
              type='text'
              placeholder='Username'
              required
              id='RegistrationForm__username'
            ></input>
            <label htmlFor='password'></label>
            <input
              type='password'
              name='password'
              placeholder='Password'
              required
              id='RegistrationForm__password'
            ></input>
          </fieldset>
          <button className='registration-button' type='submit'>
            Register
          </button>
        </form>
      </>
    );
  }
}
