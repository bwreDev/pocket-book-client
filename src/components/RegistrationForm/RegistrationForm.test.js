import React from 'react';
import ReactDOM from 'react-dom';
import RegistrationForm from './RegistrationForm';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <RegistrationForm />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <RegistrationForm />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
