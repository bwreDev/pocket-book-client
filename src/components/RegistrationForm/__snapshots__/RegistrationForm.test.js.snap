// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <nav
    className="navbar"
    role="navigation"
  >
    <ul
      className="navbar-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>,
  <form
    className="RegistrationForm"
    onSubmit={[Function]}
  >
    <fieldset
      className="Registration-input"
    >
      <h3>
        Register
      </h3>
      <label
        htmlFor="RegistrationForm__first_name"
      />
      <input
        id="RegistrationForm__first_name"
        name="first_name"
        placeholder="First Name"
        required={true}
        type="text"
      />
      <label
        htmlFor="RegistrationForm__last_name"
      />
      <input
        id="RegistrationForm__last_name"
        name="last_name"
        placeholder="Last Name"
        required={true}
        type="text"
      />
      <label
        htmlFor="RegistrationForm__username"
      />
      <input
        id="RegistrationForm__username"
        name="username"
        placeholder="Username"
        required={true}
        type="text"
      />
      <label
        htmlFor="password"
      />
      <input
        id="RegistrationForm__password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
    </fieldset>
    <button
      className="registration-button"
      type="submit"
    >
      Register
    </button>
  </form>,
]
`;
